[2024-05-01T00:41:03.455+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-05-01T00:41:03.471+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: batch_analysis_dag.test_query manual__2024-05-01T00:41:01.729131+00:00 [queued]>
[2024-05-01T00:41:03.477+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: batch_analysis_dag.test_query manual__2024-05-01T00:41:01.729131+00:00 [queued]>
[2024-05-01T00:41:03.477+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 1
[2024-05-01T00:41:03.486+0000] {taskinstance.py:2327} INFO - Executing <Task(_PythonDecoratedOperator): test_query> on 2024-05-01 00:41:01.729131+00:00
[2024-05-01T00:41:03.495+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=605) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-05-01T00:41:03.496+0000] {standard_task_runner.py:63} INFO - Started process 607 to run task
[2024-05-01T00:41:03.496+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'batch_analysis_dag', 'test_query', 'manual__2024-05-01T00:41:01.729131+00:00', '--job-id', '53', '--raw', '--subdir', 'DAGS_FOLDER/ml-dag.py', '--cfg-path', '/tmp/tmp5fk6m4v1']
[2024-05-01T00:41:03.498+0000] {standard_task_runner.py:91} INFO - Job 53: Subtask test_query
[2024-05-01T00:41:03.530+0000] {task_command.py:426} INFO - Running <TaskInstance: batch_analysis_dag.test_query manual__2024-05-01T00:41:01.729131+00:00 [running]> on host 0a71bd2d4f31
[2024-05-01T00:41:03.586+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='batch_analysis_dag' AIRFLOW_CTX_TASK_ID='test_query' AIRFLOW_CTX_EXECUTION_DATE='2024-05-01T00:41:01.729131+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-05-01T00:41:01.729131+00:00'
[2024-05-01T00:41:03.587+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-05-01T00:41:03.593+0000] {base.py:84} INFO - Using connection ID 'test-mongo-con' for task execution.
[2024-05-01T00:41:03.601+0000] {python.py:237} INFO - Done. Returned value was: <pymongo.cursor.Cursor object at 0x7fd00ec23410>
[2024-05-01T00:41:03.602+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-05-01T00:41:03.608+0000] {xcom.py:677} ERROR - Object of type Cursor is not JSON serializable. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your *** config or make sure to decorate your object with attr.
[2024-05-01T00:41:03.609+0000] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 91, in default
    return serialize(o)
           ^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/serialization/serde.py", line 189, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'pymongo.cursor.Cursor'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 486, in _execute_task
    task_instance.xcom_push(key=XCOM_RETURN_KEY, value=xcom_value, session=session)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in xcom_push
    XCom.set(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/xcom.py", line 247, in set
    value = cls.serialize_value(
            ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/xcom.py", line 675, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 104, in encode
    return super().encode(o)
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 93, in default
    return super().default(o)
           ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type Cursor is not JSON serializable
[2024-05-01T00:41:03.616+0000] {taskinstance.py:1205} INFO - Marking task as FAILED. dag_id=batch_analysis_dag, task_id=test_query, execution_date=20240501T004101, start_date=20240501T004103, end_date=20240501T004103
[2024-05-01T00:41:03.624+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 53 for task test_query (Object of type Cursor is not JSON serializable; 607)
[2024-05-01T00:41:03.669+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-05-01T00:41:03.682+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-05-01T00:41:03.684+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
